### DISCUSSION TITLE
Problem with root uri containing and extra level.

### DISCUSSION THREAD

**Discussion by lsces** [nginx/nginx#203] [Q&A] [answered]
I have a web application that consists of several modules each in it's own sub-directory off the root directory. /wiki/, /articles/, /blogs/ etc. It has been some time since I had this last working as I have been out of circulation and am only now returning these sites to active. In addition, it is reliant on PHP7.3, as later code changes in PHP7.4 need to be repaired. SO I missed a problem initially where Nginx needed a 'root' added one level up in the config file. I assumed the problem was with the PHP side, but now I am slowly getting back up to speed I'm remembering just where to check.
The basic problem I'm currently fighting is - I think - that rather than using 'localhost' as the root uri, Nginx is picking up the /wiki/ and adding that, so that switching modules fails. localhost/wiki/xxx has pages in localhost/articles/xxx but Nginx is redirecting to localhost/wiki/articles/xxx which obviously fails. Checking old log files I think confirms that this is a server problem and the files being produced are correctly addressing the different modules. 
I HAVE succeeded in getting the setup to display pictures which was the first problem and adding the extra root was all that was needed.
[nginx.txt](https://github.com/user-attachments/files/17090133/nginx.txt)
Copy of the current .conf file ... adding a ./ to links to strip the extra level works, but causes other problems when the extra level is missing.

**Comment by lsces** [comment, marked as answer]
OK, don't ask me what has changed over night, but I rolled back to the first site I had been playing with and it just worked perfectly. Switched back to the one I was trying to debug and THAT is working fine as well. The nginx.conf file above is wrong as I'd disabled the @rewrite element while debugging. Time now to sort out 5 year old crib sheets ...
One thing I DO have now in the current .conf is a single root entry in the top level of the server section and now all I need to do is change that one entry switch sites until I can get an external setup pointed by the dns for the domains ... and switching the fastcgi_pass port selects the right version of PHP. Just need now to convince openSUSE that I can run PHP8 in parallel with PHP7.3 and PHP7.4 ...

**Comment by Maryna-f5** [comment]
Thank you, @lsces ! Closing this discussion.
